src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -204,15 +206,25 @@ private static bool TryParseStatusLine(HttpResponseMessage response, string resp
                 return false;
             }
 
+	    //Clear the header if status line is recieved again. This signifies that there are multiple response headers (like in redirection).
+
+            response.Headers.Clear();
+            
+            response.Content.Headers.Clear();        
+
+            int responseHeaderLength = responseHeader.Length;
+
             // Check if line begins with HTTP/1.1 or HTTP/1.0
             int prefixLength = s_httpPrefix.Length;
-            if ((responseHeader[prefixLength] == '1') && (responseHeader[prefixLength + 1] == '.'))
+            int versionIndex = prefixLength + 2;
+
+            if ((versionIndex < responseHeaderLength) && (responseHeader[prefixLength] == '1') && (responseHeader[prefixLength + 1] == '.'))
             {
-                if (responseHeader[prefixLength + 2] == '1')
+                if (responseHeader[versionIndex] == '1')
                 {
                     response.Version = new Version(1, 1);
                 }
-                else if (responseHeader[prefixLength + 2] == '0')
+                else if (responseHeader[versionIndex] == '0')
                 {
                     response.Version = new Version(1, 0);
```

Comment:
The HttpVersion class is INTERNAL only for CoreFx (although it was public in .NET Desktop).

WinHttpHandler code could be changed to reference the internal static fields of HttpVersion but it wasn't at the time since the source code for that class wasn't in the Common/src/System/Net folder tree.  But that optimization could be done at a future time.  I agree that it would save having to do a `new Version(...)` allocation each time.